lib/pages/note_page.dart

```diff
@@ -1,19 +1,176 @@
 import 'package:flutter/material.dart';
+import 'package:soccer_app_flutter/pages/note_page/color_setting_page.dart';
+import 'package:soccer_app_flutter/shared/enums/navigation_items.dart';
+import 'package:soccer_app_flutter/pages/main_navigation_bar.dart';
 
-// NotePage（自由帳ページ）
-class NotePage extends StatelessWidget {
+class NotePage extends StatefulWidget {
   const NotePage({super.key});
 
+  @override
+  State<NotePage> createState() => _NotePageState();
+}
+
+class _NotePageState extends State<NotePage> {
+  final _formKey = GlobalKey<FormState>();
+
+  String _title = '';
+  String _description = '';
+  String _category = '';
+  String _difficulty = '初級';
+  int _phase = 1;
+  int _ledCount = 1;
+
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       key: const Key("notePage"),
       appBar: AppBar(
         backgroundColor: Theme.of(context).colorScheme.inversePrimary,
-        title: const Text('自由帳 デモページ'),
+        title: const Text('自由帳'),
       ),
-      body: const Center(
-        child: Text('このページに自由帳の内容を記載予定', style: TextStyle(fontSize: 24)),
+      body: Padding(
+        padding: const EdgeInsets.all(16.0),
+        child: Form(
+          key: _formKey,
+          child: ListView(
+            children: [
+              TextFormField(
+                decoration: const InputDecoration(
+                  labelText: 'タイトル（20字以内）',
+                  border: OutlineInputBorder(),
+                ),
+                maxLength: 20,
+                onSaved: (value) => _title = value ?? '',
```

Comment:
copyWithメソッドを使うとこんな感じで書けそうです
```suggestion
                onSaved: (value) => practiceMenu = practiceMenu.copyWith(name: value ?? ''),
```

Reply:
フォームの入力内容はローカル変数で管理して最後に1回だけオブジェクトを作成する感じにしようと思います